examples/run_CPMG.py

```diff
@@ -31,31 +30,25 @@
 # {{{set phase cycling
 phase_cycling = True
 if phase_cycling:
-    ph1_cyc = r_[0, 1, 2, 3]
-    nPhaseSteps = 4
+    ph1_cyc = array([(j*2+k)%4 for k in range(4) for j in range(2)])
+    ph2_cyc = array([(k+1)%4 for k in range(4) for j in range(2)])
+    nPhaseSteps = 8
+
 if not phase_cycling:
     ph1_cyc = 0.0
     nPhaseSteps = 1
 # }}}
-# {{{better tau
-marker = 1.0  # 10/10/22 → what is this? → pretty sure the time needed to execute the marker command
-pad_start = config_dict["tau_extra_us"] - config_dict["deadtime_us"]
-pad_end = (
-    config_dict["tau_extra_us"] - config_dict["deblank_us"] - marker
-)  # marker + deblank
-assert(pad_start > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-assert (pad_end > 0), "tau_extra_us must be set to more than deadtime and more than deblank!"
-twice_tau_echo_us = (  # the period between 180 pulses
-    config_dict["tau_extra_us"] * 2 + config_dict["acq_time_ms"] * 1e3
+# {{{symmetric tau
+marker = 1.0
+jumpto = 1.0
+pad_end = config_dict['deadtime_us'] - config_dict['deblank_us'] - marker - jumpto
+twice_tau_echo_us = (  # the period between end of first 180 pulse and start of next
+    config_dict['deadtime_us'] + config_dict["acq_time_ms"] * 1e3 + pad_end + marker + config_dict['deblank_us']
```

Comment:
- are you sure that you are rounding acq_time_ms to an integral number of dwell times? (I have written code to do this, so it should be available somewhere).  That needs to be done in this pulse sequence.  If so, you should add a comment where that is done, referencing that it's necessary to make this work correctly.
- base on previous, why not just multiply deadtime by 2? (Since you are forcing the stuff on the right to be equal to the deadtime.

Reply:
line 24 handles the rounding acq_time

Reply:
fixed twice tau to be 2xdeadtime +acq